### PR TITLE
vmalert: allow configuring custom notifier headers per group

### PR DESCRIPTION
**Pull request by Haleygo**
fix https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3260

### CODE REVIEW

**Comment by Haleygo** [comment]
@hagen1778 could you help reviewing this?
I suppose there is no need to support label template in headers for now :)

**Comment by Haleygo** [comment]
@hagen1778 PTAL :)

**Comment by hagen1778** [comment]
Hi @Haleygo! I'll take a look once I have a chance.
Just returned from 3w vacation and catching up with everything I missed.

File: app/vmalert/group.go [hunk 1]
```diff
@@ -134,15 +139,15 @@ func newGroup(cfg config.Group, qb datasource.QuerierBuilder, defaultInterval ti
...
 	}
 	g.Rules = rules
 	return g
 }
 
-func (g *Group) newRule(qb datasource.QuerierBuilder, rule config.Rule) Rule {
+func (g *Group) newRule(qb datasource.QuerierBuilder, rule config.Rule, notifierHeaders map[string]string) Rule {
 	if rule.Alert != "" {
-		return newAlertingRule(qb, g, rule)
+		return newAlertingRule(qb, g, rule, notifierHeaders)
```
[the remark this answers is not in the document]
Comment (Haleygo) [reply app/vmalert/group.go]: Good point, since each group create its own executor, I will set header in there.

**Review by hagen1778 — APPROVED** [review]
Thank you!

**Comment by valyala** [comment]
FYI, this change has been included in `vmalert` [v1.91.0](https://docs.victoriametrics.com/CHANGELOG.html).

**Pull request merged**